Refactor Pipeline

valendesigns opened the pull request:
- [x] Verify the API is available to all Services
- [x] Test Sync Server
- [x] Test PHPCS Server
- [x] Test Lighthouse Server

close #28, closes #32, closes #38

rheinardkorf committed:
Refactored Lighthouse service to use new pipeline processes.

rheinardkorf committed:
Replace phpcs service with new pipeline processes.

rheinardkorf committed:
glide.lock manually updated to pull pre-merged branch

rheinardkorf committed:
Refactored phpcs-server

rheinardkorf committed:
Merge branch 'develop' into refactor/pipeline

rheinardkorf committed:
Refactored phpcs-server

rheinardkorf committed:
Fix processes and update dependencies

rheinardkorf committed:
Add glide.yaml

rheinardkorf committed:
Update lighthouse-server

rheinardkorf committed:
Update glide.lock

valendesigns committed:
Merge develop

valendesigns committed:
Fix env vars in test files

rheinardkorf committed:
Merge branch 'develop' into pipeline

rheinardkorf committed:
Add Intercept processor. Not integrated into pipeline yet.

rheinardkorf committed:
Inject Intercept process into pipeline.

rheinardkorf commented:
@valendesigns de24b66 should now cover #38 ...
Can you please verify the hardcoded path value?

valendesigns committed:
Version global composer packages

valendesigns commented:
@rheinardkorf https://github.com/xwp/go-tide/pull/48/commits/2348b0db9d23a9624566cb5a2b14931a045e62bf ensures the file is at that location. We'll need to update to a specific version once the file has been merged into an actual tag.

valendesigns committed:
Ignore paths in overalls coverage

rheinardkorf commented:
@valendesigns nice!

rheinardkorf committed:
Fix nil interface reference.

rheinardkorf committed:
Merge branch 'pipeline' of github.com:xwp/go-tide into pipeline

rheinardkorf committed:
Update phpcs standard override

valendesigns committed:
Upgrade apk-tools in .build-deps

rheinardkorf committed:
Merge branch 'develop' into pipeline

valendesigns committed:
Update npm on the image

rheinardkorf committed:
Fix server termination bug.

rheinardkorf committed:
Merge branch 'pipeline' of github.com:xwp/go-tide into pipeline

rheinardkorf committed:
Use modified pipeline with provided error channel.

Pipeline now uses an error channel passed as an argument. This helps
simplify reading error messages from the services and eliminates the
need to merge multiple error channels.

Also added debug process for use in main services.

rheinardkorf committed:
Merge branch 'develop' into pipeline

rheinardkorf committed:
Refactored for sequential process. Needs tests

rheinardkorf committed:
Fix test coverage, update glide.locl

rheinardkorf committed:
Update PHPCompatibilityWPPath

rheinardkorf committed:
Add -wp-phpcompat-ruleset flag.

valendesigns commented on cmd/phpcs-server/testdata/wordpress.xml:
```diff
@@ -0,0 +1,46 @@
+<ruleset name="PHPCompatibility-WP">
+    <description>WordPress specific ruleset which checks for PHP cross version compatibility.</description>
+
+    <autoload>./../PHPCSAliases.php</autoload>
```
Wouldn't this now be a broken path?

rheinardkorf replied:
it would, the point is not so much what is in the file... just that the file is indeed loaded if it is a `theme` or `plugin` project

valendesigns commented on cmd/phpcs-server/main_test.go:
```diff
@@ -135,6 +136,14 @@ func Test_main(t *testing.T) {
+			args{
+				messageChannel: make(chan message.Message, 1),
+				timeOut:        1,
+				flagWPRuleset:  &[]string{"./testdata/wordpress.xml"}[0],
```
Why not just use the actual standard in place of the ruleset file and then we don't have to include it in the testdata directory?

valendesigns replied:
I don't think we need to test that the file works, it's assumed a ruleset can be passed to the cli because it's a feature of `phpcs`. What is the benefit of storing the ruleset in `testdata`?

rheinardkorf replied:
We don't need to...  unless we want coverage, because of the new cli flag.
We store the file in `testdata` because that is where the tests have access to the file. As per a previous comment, its not so much what is in it, rather that it loads instead of the normal standard.  So we don't need to keep this in sync with the real file.

valendesigns replied:
Or we could just make it a ruleset that loads the PHPCompatibility standard to verify the file is used. Then no need to keep it in sync.

valendesigns committed:
Update wptide/pkg

rheinardkorf committed:
Add PayloadType to Sync Server

rheinardkorf committed:
Update glide.lock

valendesigns committed:
Update wp-tide-api

valendesigns committed:
Update wptide/pkg

rheinardkorf committed:
Use 'tide' for sync-server payload type

valendesigns committed:
Update wptide/pkg
